src/assets/scripts/client/aircraft/StripView/StripViewController.js

```diff
@@ -257,7 +277,53 @@ export default class StripViewController {
      * @private
      */
     // eslint-disable-next-line no-unused-vars
-    _onStripListClickOutsideStripViewModel = (event) => {
-        this.findAndDeselectActiveStripView();
-    };
+    _onStripListClickOutsideStripViewModel = (event) => this.findAndDeselectActiveStripView();
+
+    /**
+     * Generate a unique number to represent a `CID`
+     *
+     * Should be displayed with leading zeros, so a `CID` value of `1` should be displayed as `001`
+     *
+     * @for StripViewController
+     * @method _generateCidNumber
+     * @return nextCid {number}
+     * @private
+     */
+    _generateCidNumber() {
+        const nextCid = _random(1, CID_UPPER_BOUND);
+        const nextCidLeft = nextCid - 1;
+        const nextCidRight = nextCid + 1;
+
+        if (
+            this._cidNumbersInUse.indexOf(nextCid) !== -1 ||
+            this._cidNumbersInUse.indexOf(nextCidLeft) !== -1 ||
+            this._cidNumbersInUse.indexOf(nextCidRight) !== -1
+        ) {
+            this._generateCidNumber();
```

Comment:
you said non-sequential CID numbers.  This if block looks for our random number `n` and then additionally `n - 1` and `n + 1`.  If any of those three exist, we re-generate because we've found either a duplicate, or a set of sequential numbers.